src/lib.rs

```diff
@@ -2,78 +2,36 @@
 
 extern crate num;
 
+mod algorithm;
 mod butterflies;
-mod mixed_radix;
-mod radix4;
-mod raders_algorithm;
 mod math_utils;
+mod array_utils;
+mod plan;
 
 use num::{Complex, Zero, One, Float, FromPrimitive, Signed};
 use num::traits::cast;
 use std::f32;
 
-use mixed_radix::cooley_tukey;
-use radix4::process_radix4;
-use raders_algorithm::RadersAlgorithm;
-
-enum Algorithm<T> {
-    MixedRadix(Vec<(usize, usize)>, Vec<Complex<T>>),
-    Radix4,
-    Raders(RadersAlgorithm<T>),
-    Noop,
-}
+use algorithm::FFTAlgorithm;
 
 pub struct FFT<T> {
-    algorithm: Algorithm<T>,
-    twiddles: Vec<Complex<T>>,
-    inverse: bool,
+    len: usize,
+    algorithm: Box<FFTAlgorithm<T>>,
 }
 
-impl<T> FFT<T> where T: Signed + FromPrimitive + Copy {
+impl<T> FFT<T>
+    where T: Signed + FromPrimitive + Copy + 'static
```

Comment:
Because of the FFTAlgorithm trait objects -- without it, the compiler is worried that T may be a type with a bounded lifetime. But any float or bigdecmal type is going to have a static lifetime, so the static bound just soothes the compiler without restricting anything.

It'd be nice to get rid of it, but I'm not sure how.